internal/datastore/postgres/gc.go

```diff
@@ -42,15 +53,32 @@ func (pgd *pgDatastore) ResetGCCompleted() {
 	pgd.gcHasRun.Store(false)
 }
 
-func (pgd *pgDatastore) Now(ctx context.Context) (time.Time, error) {
+func (pgg *pgGarbageCollector) Close() {
+	pgg.isClosed = true
+	pgg.conn.Release()
+}
+
+func (pgg *pgGarbageCollector) LockForGCRun(ctx context.Context) (bool, error) {
+	return pgg.pgd.tryAcquireLock(ctx, pgg.conn, gcRunLock)
```

Comment:
I think the locking could probably move in to the garbage collector too?

Reply:
We use it for non-GC too